python/paddle/distributed/auto_parallel/tuner/algorithms.py

```diff
@@ -0,0 +1,159 @@
+#   Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
```

Comment:
It may be better to make all the tuners inherited from Tuner base class. This algorithm introduce another indirection and abstract concept. You can improve the Tuner base class. The optimization tuner is a composite tuner, which can call the sharding tuner and the recompute tuner.

Reply:
will be considered